src/materials/ConcreteThermalMoisture.C

```diff
@@ -382,206 +400,212 @@ ConcreteThermalMoisture::computeProperties()
+        if (_water_to_cement < 0.3)
+          N_wc = 0.99;
+        else if (_water_to_cement >= 0.3 && _water_to_cement <= 0.7)
+          N_wc = 0.33 + 2.2 * _water_to_cement;
+        else
+          N_wc = 1.87;
 
-    Real Vm = V_te * V_wc * V_ct * 1.0;
+        Real N_te = 5.5;
+        if (_eqv_age[qp] >= 5)
+          N_te = 2.5 + 15.0 / _eqv_age[qp];
 
-    Real W_cement = C * k * Vm * H / (1.0 - k * H) / (1.0 + (C - 1.0) * k * H);
+        Real n = N_te * N_wc * N_ct * 1.0;
 
-    Real dWdH_cement = (C * k * Vm + W_cement * k * (1.0 + (C - 1.0) * k * H) -
-                        W_cement * k * (1.0 - k * H) * (C - 1.0)) /
-                       (1.0 - k * H) / (1.0 + (C - 1.0) * k * H);
+        Real k = ((1.0 - 1.0 / n) * C - 1.0) / (C - 1.0);
+        if (k < 0.0)
+          k = 0.0;
+        else if (k > 1.0)
+          k = 1.0;
 
-    // compute mositure capacity dw/dH for aggregates
-    Real n_agg = 1.5;
-    switch (_aggregate_pore_type)
-    {
-      case 0:
-        n_agg = 1.5;
-        break;
-      case 1:
-        n_agg = 2.0;
-        break;
-      default:
-        mooseError("Unknown aggregate pore structure");
-        break;
-    }
-
-    n = 4.063 * n_agg;
-    k = ((1.0 - 1.0 / n) * C - 1.0) / (C - 1.0);
-    if (k < 0.0)
-      k = 0.0;
-    else if (k > 1.0)
-      k = 1.0;
-
-    Real V_agg = 0.05;
-    switch (_aggregate_pore_type)
-    {
-      case 0:
-        V_agg = 0.05;
-        break;
-      case 1:
-        V_agg = 0.10;
-        break;
-      default:
-        mooseError("Unknown aggregate pore structure");
-        break;
-    }
+        Real V_ct = 0.9;
+        switch (_cement_type)
+        {
+          case 0:
+            V_ct = 0.9;
+            break;
+          case 1:
+            V_ct = 1.0;
+            break;
+          case 2:
+            V_ct = 0.85;
+            break;
+          case 3:
+            V_ct = 0.6;
+            break;
+          default:
+            mooseError("Unknown cement type in mositure capacity calculations");
+            break;
+        }
 
-    Vm = 0.0647 * V_agg;
+        Real V_wc = 0.985;
+        if (_water_to_cement < 0.3)
+          V_wc = 0.985;
+        else if (_water_to_cement >= 0.3 && _water_to_cement <= 0.7)
+          V_wc = 0.85 + 0.45 * _water_to_cement;
+        else
+          V_wc = 1.165;
 
-    Real W_agg = C * k * Vm * H / (1.0 - k * H) / (1.0 + (C - 1.0) * k * H);
-    Real dWdH_agg = (C * k * Vm + W_agg * k * (1.0 + (C - 1.0) * k * H) -
-                     W_agg * k * (1.0 - k * H) * (C - 1.0)) /
-                    (1.0 - k * H) / (1.0 + (C - 1.0) * k * H);
+        Real V_te = 0.024;
+        if (_eqv_age[qp] >= 5)
+          V_te = 0.068 - 0.22 / _eqv_age[qp];
 
-    // compute weight percentages of agrregates and cement paste
-    Real _f_agg = _aggregate_mass / (_aggregate_mass + _cement_mass);
-    Real _f_cp = _cement_mass / (_aggregate_mass + _cement_mass);
+        Real Vm = V_te * V_wc * V_ct * 1.0;
 
```

Comment:
Why is there a factor 1.0?

Reply:
This factor is to account for the influence of temperature in the calculation of monolayer capacity. 

I have mentioned this comment in the source file now.

Reply:
This would be better handled by doing something like this:
```
const Real monolayer_factor = 1.0;
Real Vm = ... * monolayer_factor;